apps/server/src/shared/domain/entity/coursegroup.entity.ts

```diff
@@ -42,14 +41,12 @@ export class CourseGroup extends BaseEntityWithTimestamps implements IEntityWith
 	}
 
 	public getStudentIds(): EntityId[] {
-		if (!this.students) {
-			throw new InternalServerErrorException(
-				'Coursegroup.students is undefined. The coursegroup need to be populated.'
-			);
-		}
+		let studentIds: EntityId[] = [];
 
-		const studentObjectIds = this.students.getIdentifiers('_id');
-		const studentIds = studentObjectIds.map((id): string => id.toString());
+		if (this.students) {
```

Comment:
Theoretisch würde ich hier für ein logger.warn plädieren. Bei DOs fänd ich das dann auch ok. Da man dort auch den logger injecten kann. Bei MikroORM Entitys wird der constructor nur von uns verwendet, sollte daher auch gehen, aber haben wir bisher nirgends gemacht.

Reply:
Also die Condition flippen Z.47+48 ins else und das warn in das if?

Reply:
Oder einfach so: 

```js
public getStudentIds(): EntityId[] {
		let studentIds: EntityId[] = [];

		if (this.students) {
			const studentObjectIds = this.students.getIdentifiers('_id');
			studentIds = studentObjectIds.map((id): string => id.toString());
		} else {
			this.logger.warn('Course group students are not populated');
		}

		return studentIds;
	}
```

Reply:
Ich bin mir unsicher ob wir injects an die Entitäten hängen können im constructor.
Daher lassen wir das am besten einfach erstmal. :D

Reply:
Wollen wir an die Stelle einen Kommentar ranschreiben? Sowas wie "Unlike in Course entity we don't throw an exception if students is not populated because of an issue with the deletion of CourseGroups: If a CourseGroup is deleted the reference in Submission is not and thus an incorrect state can arise. We can't fix this at the moment and to handle it decently we only log here." 

Ansonsten kann man den Unterschied nicht nachvollziehen, finde ich.